staubli_val3_driver/val3/ros_server/start.pgx

```diff
@@ -1,141 +1,133 @@
-
-  // create joint state and robot status streamer task to send data to industrial_robot_client
-  taskCreateSync sStreamTaskName, l_nFbkPeriod, bFbkOverrun, dataStreamer()
-  
-  // create trajectory point task to receive data from industrial_robot_client
-  taskCreateSync sTrajPtTaskName, l_nTrajPeriod, bTrajPtOverrun, dataReceiver()
-  //taskCreate sTrajPtTaskName, 100, dataReceiver()
-  
-  // create motion task to synchronously apply motion commands to robot
-  taskCreateSync sMotionTaskName, l_nMotionPeriod, bMotionOverrun, motionControl()
-
-  // create synchronous task to update screen
-  //taskCreateSync sScreenTaskName, l_nScreenUpdate, bScreenOverrun, screenUpdate()
-  taskCreate sScreenTaskName, 10, screenUpdate()
-  
-  // create asynchronous task to receive heartbeat on port 11002
-  taskCreate sHtbtTaskName, 10, recvHeartbeat()
-
-end]]></Code>
-  </Program>
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Programs xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://www.staubli.com/robotics/VAL3/Program/2">
+  <Program name="start">
+    <Locals>
+      <Local name="l_nScreenUpdate" type="num" xsi:type="array" size="1" />
+      <Local name="l_sVersionMajor" type="string" xsi:type="array" size="1" />
+      <Local name="l_nVersionMajor" type="num" xsi:type="array" size="1" />
+      <Local name="l_bOkMajor" type="bool" xsi:type="array" size="1" />
+      <Local name="l_sVersionMinor" type="string" xsi:type="array" size="1" />
+      <Local name="l_nVersionMinor" type="num" xsi:type="array" size="1" />
+      <Local name="l_bOkMinor" type="bool" xsi:type="array" size="1" />
+      <Local name="l_nLibLoad" type="num" xsi:type="array" size="1" />
+    </Locals>
+    <Code><![CDATA[begin
+  // Copyright (c) 2016, Ocado Technology - Robotics Research Team
+  //
+  // Licensed under the Apache License, Version 2.0 (the "License");
+  // you may not use this file except in compliance with the License.
+  // You may obtain a copy of the License at
+  //
+  //     http://www.apache.org/licenses/LICENSE-2.0
+  //
+  // Unless required by applicable law or agreed to in writing, software
+  // distributed under the License is distributed on an "AS IS" BASIS,
+  // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  // See the License for the specific language governing permissions and
+  // limitations under the License.
+
+
+  // Load interface library for CS8 or CS9
+  //
+  // CS9 handles the UI differently than CS8, requires
+  // the UI funtions to be loaded in seperate libraries
+  
+  if(mid(getVersion("System"),1,1) == "7")
+    // CS8 - VAL 3 rev 7
+    // Check for VAL 3 version >= 7.7
+    l_sVersionMajor = mid(getVersion("VAL3"),1,1)
+    l_sVersionMinor = mid(getVersion("VAL3"),2,1)
+    toNum(l_sVersionMajor, l_nVersionMajor, l_bOkMajor)
+    toNum(l_sVersionMinor, l_nVersionMinor, l_bOkMinor)
+    if(!l_bOkMajor or !l_bOkMinor)
+      popUpMsg("Error reading VAL 3 version")
+      call stop()

```

Comment:
I would do this, calling the `stop()` here will run the stop program, and because after this the application stops running, the `stop()` program is called again. I would just end the start program either by doing nothing or with a `return` otherwise the `stop()` program is running twice!